src/temporary_home/ResumeCache.cpp

```diff
@@ -132,6 +132,33 @@ bool ResumeCache::isPositionAvailable(
   return result;
 }
 
+StreamId ResumeCache::getStreamId(const std::string& streamName) const {
+  /// This is a little inefficient.  But is called very infrequently
+  for (const auto& it : streamNameMap_) {
+    if (it.second == streamName) {
+      return it.first;
+    }
+  }
+  return 0;
+}
+
+std::unordered_map<StreamId, std::string /* streamName */>
```

Comment:
This can probably be `std::vector<Stream>` or something as order doesn't seem important. Gets rid of the `std::pair` name problem too.